test/versioned/fastify/add-hook.tap.js

```diff
@@ -0,0 +1,83 @@
+/*
```

Comment:
i know this is not obvious, but you have to add this test to the `test/versioned/fastify/package.json` for each test stanza.  It looks like it does not work in v2 and am getting failures in v3

```
        not ok 2 - segment "WebTransaction/WebFrameworkUri/Fastify/GET//" should have child "Nodejs/Middleware/Fastify/preValidation/testHook" in position 3: expected 'Nodejs/Middleware/Fastify/onSend/testHook' to equal 'Nodejs/Middleware/Fastify/preValidation/testHook'
```

Reply:
looking at the trace root there are only 3 of the 7 handlers firing

```
in lifecycle event onRequest
in lifecycle event preParsing
in lifecycle event onSend
```

Reply:
looks like maybe because the signatures vary https://www.fastify.io/docs/latest/Hooks/.  Also I do see there are more hooks in there that I didn't previously call out like `onTimeout` and `onError` but I don't see them in the request lifecycle tree.